src/api/v4/dataview/category.js

```diff
@@ -0,0 +1,241 @@
+     * @property {string} operation
+     * @property {CategoryItem[]} result
+     * @property {string} type - Constant 'category'
+     * @api
+     */
+    var data = this._internalModel.get('data');
+    var result = _.map(data, function (item) {
+      return {
+        group: item.agg,
+        name: item.name,
+        value: item.value
+      };
+    });
+    return {
+      count: this._internalModel.get('count'),
+      max: this._internalModel.get('max'),
+      min: this._internalModel.get('min'),
+      nulls: this._internalModel.get('nulls'),
+      operation: this._options.operation,
+      result: result,
+      type: 'category'
+    };
+  }
+  return null;
+};
+
+Category.prototype.DEFAULTS = {
+  maxCategories: 6,
+  operation: constants.operation.COUNT,
+  operationColumn: 'column'
+};
+
+Category.prototype._listenToInternalModelSpecificEvents = function () {
+  this.listenTo(this._internalModel, 'change:categories', this._onMaxCategoriesChanged);
+  this.listenTo(this._internalModel, 'change:aggregation', this._onOperationChanged);
+  this.listenTo(this._internalModel, 'change:aggregation_column', this._onOperationColumnChanged);
+};
+
+Category.prototype._onMaxCategoriesChanged = function () {
+  if (this._internalModel) {
+    this._options.maxCategories = this._internalModel.get('categories');
+  }
+  this.trigger('maxCategoriesChanged', this._options.maxCategories);
+};
+
+Category.prototype._onOperationChanged = function () {
+  if (this._internalModel) {
+    this._options.operation = this._internalModel.get('aggregation');
+  }
+  this.trigger('operationChanged', this._options.operation);
+};
+
+Category.prototype._onOperationColumnChanged = function () {
+  if (this._internalModel) {
+    this._options.operationColumn = this._internalModel.get('aggregation_column');
+  }
+  this.trigger('operationColumnChanged', this._options.operationColumn);
+};
+
+Category.prototype._checkOptions = function (options) {
+  if (_.isUndefined(options)) {
+    throw new TypeError('Category dataview options are not defined.');
+  }
+  this._checkMaxCategories(options.maxCategories);
+  this._checkOperation(options.operation);
+  this._checkOperationColumn(options.operationColumn);
+};
+
+Category.prototype._checkMaxCategories = function (maxCategories) {
+  if (_.isUndefined(maxCategories)) {
+    throw new TypeError('Max categories for category dataview is required.');
+  }
+  if (!_.isNumber(maxCategories)) {
+    throw new TypeError('Max categories for category dataview must be a number.');
+  }
+  if (maxCategories <= 0) {
+    throw new TypeError('Max categories for category dataview must be greater than 0.');
+  }
+};
+
+Category.prototype._checkOperation = function (operation) {
+  if (_.isUndefined(operation) || !constants.isValidOperation(operation)) {
+    throw new TypeError('Operation for category dataview is not valid. Use carto.operation');
+  }
+};
+
+Category.prototype._checkOperationColumn = function (operationColumn) {
```

Comment:
Do all operations support `operationColumn`? If that's not the case... should we throw an error?

Reply:
All operations can have an operationColumn.

Depending on the column type, it can throw an error, but it's Maps API the one that throws it.